src/Rotations.jl

```diff
@@ -1,5 +1,47 @@
+# we generate code in this module, so precompile where possible
+ VERSION >= v"0.4.0-dev+6521" && __precompile__(true)
+
 module Rotations
 
-# package code goes here
+using FixedSizeArrays
+using Quaternions  # TODO: Switch to the UnitQuaternions package? https://github.com/kam3k/UnitQuaternions.jl
+
+import Base: convert, mean, eltype, isnan, length, call, getindex, inv, .*, tuple
+import Quaternions.Quaternion
+
+src_files = ["rotation_types", "rotation_conversions", "euler_conversions", "type_methods", "rotate", "quaternion_ops", "math_funcs", "rot_covariances"]
+for file in src_files
+    #println("$(file).jl")
+    include("$(file).jl")
+end
+
+
+export
+
+        # Euler angle ordering
+        EulerOrder,
+        EulerZXY,
+
+        # representations
+        RotMatrix,
+        SpQuat,
+        AngleAxis,
+        EulerAngles,         # order is in the type specification
+        ProperEulerAngles,   # order is in the type specification
```

Comment:
The distinction between `EulerAngles` and `ProperEulerAngles` seems a little unnecessary because all the ordering information is held in the `EulerOrder` anyway.  However I can see that for `EulerAngles` there's a fairly straightforward API to access the individual rotations (since x,y,z rotations are always present in some order).  For the `ProperEulerAngles` it's a bit more tricky since a given axis repeats and one of the axes doesn't appear at all.

I think it's premature to worry overly much about this now - let's just note that they could be put together in an issue (if it makes sense at all), and we can refactor later.